src/Simulation/CsharpGeneration.Tests/SimulationCodeTests.fs

```diff
@@ -3685,10 +3685,21 @@ namespace Microsoft.Quantum.Tests.UnitTests
                     baseSim.OnLog += this.Output.WriteLine;
                 }
 
-                sim.Run<UnitTest1, QVoid, QVoid>(QVoid.Instance).Wait();
-                if (sim is IDisposable disposeSim)
+                try
+                {
+                    sim.Execute<UnitTest1, QVoid, QVoid>(QVoid.Instance);
+                }
+                catch (Exception e)
+                {
+#line 22 "%%"
```

Comment:
What are these `#line 22 %%` markers for? I'm not familiar with that syntax.

Reply:
This is so that the error message that gets throw is assigned a line number and source file position that is relevant to the Q# source code, rather than this C# generated code. The error will appear to come from the Q# because of this, rather than this C#, which is likely more useful to Q# developers who are debugging the error.

Reply:
Oh, that's really cool. Thanks for explaining!